kbi/0018/index.rst

```diff
@@ -0,0 +1,194 @@
+
+  $ datalad siblings -d bar
+  .: here(+) [git]
+  .: origin(-) [ssh://example.com/path/to/store/7df/cc05d-b7ba-4b32-b7b0-9f9bb6edcf9d (git)]
+  .: ria-store-storage(+) [ora]
+
+The top level dataset also has its ``*-storage`` remote enabled automatically, but since it was cloned from GitHub, its origin remote points there::
+
+  $ datalad siblings -d .
+  .: here(+) [git]
+  .: origin(-) [<github repo url> (git)]
+  .: ria-store-storage(+) [ora]
+
+
+Adding RIA git remote manually
+------------------------------
+
+The dataset which was cloned directly from GitHub (superdataset in the
+example above) has GitHub as its origin. The ``ria-store-storage``
+(autoenabled git-annex special remote) is already available, but the
+git remote (formerly named ``ria-store``) is not. If we want to push
+superdatsets's git updates (not just annexed contents) back to the RIA
+store, we need to configure the git remote.
+
+There are plans to `allow adding git remotes other than origin`_
+automatically, but no implementation yet.
+
+Although it was created as part of a RIA store, the git remote is no
+different from any other git remote, and can be enabled with ``git
+remote add``. We need to know the store URL, and dataset ID. Since
+this is a git remote, we cannot use the ``ria+``, ``#{id}`` or
+``#~alias`` notation, and we have to split the ID with a path
+separator after the first three characters::
+
+   $ datalad configuration get datalad.dataset.id
+   4183e386-1fb7-467c-a508-cea7d6b1f8e6
+   $ git remote add ria-store "ssh://example.com/path/to/store/418/3e386-1fb7-467c-a508-cea7d6b1f8e6"
+
+If for some reason this step needs to be repeated for all subdatasets
+(e.g. they were installed from another source) it should be possible
+to create a short script that figures out the URL, and run it with
+``datalad foreach-dataset``.
+
+Gitmodules file
+---------------
+
+Since in our example the subdatasets were created using ``datalad
+create`` (rather than cloned into the superdataset), their urls only
+record the local path:
+
+.. code-block:: cfg
+
+  [submodule "bar"]
+        path = bar
+        url = ./bar
+        datalad-id = 7dfcc05d-b7ba-4b32-b7b0-9f9bb6edcf9d
+
+Had the superdataset been cloned from a ``ria+[http|https|ssh]`` URL,
+no source candidate configuration would be necessary, as DataLad would
+(by default) use the combination of superdataset origin and the local
+path as one of the source candidates. This would naturally not work
+when the superdataset is on GitHub, and subdatasets are not.
+
+However, the gitmodules file can be edited to contain the "right" URL,
+as it is also one of the default source candidates:
+
+.. code-block:: cfg
+
+  [submodule "bar"]
+        path = bar
+        url = ssh://example.com/path/to/store/7df/cc05d-b7ba-4b32-b7b0-9f9bb6edcf9d
+        datalad-id = 7dfcc05d-b7ba-4b32-b7b0-9f9bb6edcf9d
+        datalad-url = "ria+ssh://example.com/path/to/store#7dfcc05d-b7ba-4b32-b7b0-9f9bb6edcf9d"
+
+Reconfigure
```

Comment:
I'm wondering if this section should be removed. What is the situation that someone would need to use `create-sibling-ria --existing reconfigure` rather than one of the other options? Since this option  has lower performance and risk of altering existing configuration, maybe its better not to include it?

Reply:
I think you are right. As it stands, the section is mostly useful in highlighting that `--existing reconfigure` is not only local. This hower is already explained in the help message:

       --existing MODE
              Action to perform, if a (storage) sibling is already configured under  the
              given  name and/or a target already exists. In this case, a dataset can be
              skipped ('skip'), an existing target repository be forcefully  re-initial‐
              ized,  and  the sibling (re-)configured ('reconfigure'), or the command be
              instructed to fail ('error'). Constraints: value must be one  of  ('skip',
              'error', 'reconfigure') [Default: 'error']

and besides, I am now guessing that someone without write access would probably be greeted with an error, making it even less applicable.

I'll remove.